### PR TITLE
Automatically generate lists of lectures, modules and modules by tag

### PR DESCRIPTION
**Pull request by willu47**
Testing out some ideas - 

I have moved the presentations into posts:
- this allows us to make use of the global `site.tags` and `site.categories` frontmatter variables
- lecture file names must be prepended with a date in `YYYY-MM-DD-` format
- content of the `_presentations` collection moved into `posts`
- `collections` entry removed from `_config.yaml`

I've added three automatically generated lists
1. a list of all lectures (posts with `categories: lecture`)
2. a list of all modules (posts with `categories: module`)
3. a list of modules grouped by tags (posts with a `tags` key in the front matter, so a post with `tags: [clews, energy]` would appear twice, once under `clews` and once under `energy`

This makes the site a bit more automated and less reliant on running the Python scripts to generate the content. This way, the site also makes use of the YAML tags for structuring the content.

### CODE REVIEW

**Comment by lorenzo-sani** [comment]
Hi @willu47, your approach is interesting and seems like a good way to make the site more automatic without using too many python scripts. Probably few changes should be done since now you moved everything to posts but I don't see any major issues

File: _includes/tags_list.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
+
+{% for tag in site.tags %}
+  {% assign t = tag | first %}
+  {% assign modules = tag | last %}
+
+<h3>{{ t | downcase }}</h3>
```
Comment (lorenzo-sani) [inline _includes/tags_list.html:6]: What does this passage do? what does `| first` and `| last` mean?

Comment (willu47) [reply _includes/tags_list.html:6]: Hi Lorenzo. I believe these are Liquid filters - so they process the array of tags and store it in a variable.

Comment (willu47) [reply _includes/tags_list.html:6]: I just copied and pasted this code from some demo sites

**Comment by willu47** [comment]
> Hi @willu47, your approach is interesting and seems like a good way to make the site more automatic without using too many python scripts. Probably few changes should be done since now you moved everything to posts but I don't see any major issues

It does make things more automatic, but I think we're beginning to show the limitations of Jekyll. Firstly, treating everything as a post does allow to use the built-in tags and categories, but it presents some issues when you try to include modules in presentations. Basically, remark.js which renders the presentations processes the YAML frontmatter required by Jekyll as an empty slide - and we end up with a lot of blank slides in the presentation.

**Comment by lorenzo-sani** [comment]
> Basically, remark.js which renders the presentations processes the YAML frontmatter required by Jekyll as an empty slide - and we end up with a lot of blank slides in the presentation.

Yes, that's why in the `create_lecture` script instead of using `{% include relative ... %}` I write directly the text of the module

**Comment by lorenzo-sani** [comment]
However one problem of working like that is that if you change one module the change will not be reflected in the lectur unless you re-create the lecture

**Comment by willu47** [comment]
I agree, writing the text directly into the lecture instead of using `include` creates a lot of potential duplication of content and we then miss opportunities to update modules in all lectures from one point - instead having to update all.

**Comment by willu47** [comment]
@lorenzo-sani - I found a "hack" to remove the blank slides by looking into the documentation of remark.js which renders the presentations.  [The `exclude` property hides a slide.](https://github.com/gnab/remark/wiki/Markdown#exclude)

Bingo - this gives us seamless presentations 🥇

**Comment by lorenzo-sani** [comment]
@willu47 Very good job, I tried your code locally and it works very well. Now I could try to implement this logic in `create_lecture.py` so that it can be used to create presentations

**Comment by willu47** [comment]
Great - well, suggest we merge this if you are happy with it and then you can work on a new branch for the Python scripts.

**Comment by lorenzo-sani** [comment]
Ok, agree

**Pull request merged**